riscv/insn_macros.h

```diff
@@ -6,4 +6,6 @@
 
 #define require(x) (unlikely(!(x)) ? throw trap_illegal_instruction(insn.bits()) : (void) 0)
 
+#define software_check(x, tval) (unlikely(!(x)) ? throw trap_software_check_fault(tval) : (void) 0)
+
```

Comment:
Could we rename trap_software_check_fault to trap_software_check.

Reply:
Done
Thanks, trap_software_check is better than trap_software_check_fault in terms of spec description.